.github/workflows/build.yaml

```diff
@@ -19,12 +23,42 @@ jobs:
         with:
           install_url: https://releases.nixos.org/nix/nix-2.22.1/install
 
+      - name: Use GitHub cache
+        uses: DeterminateSystems/magic-nix-cache-action@main
+
       - name: Check Nix formatting
         run: nix fmt . -- --check
 
       - name: Check Nix flake
         run: nix flake check --all-systems
 
+  script-test:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: azohra/shell-linter@latest
+
+      - name: Test bump-holochain.sh
+        run: |
+          MINOR_VERSION=$(./scripts/bump-holochain.sh main-0.1 false | tail -n 1)
+          if [[ "$MINOR_VERSION" != "holochain-0.1.8" ]]; then
+              echo "bump.sh failed to find 0.1.8: [${MINOR_VERSION}]"
+              exit 1
+          fi
+          
+          RC_VERSION=$(./scripts/bump-holochain.sh main-0.1-rc false | tail -n 1)
+          if [[ "$RC_VERSION" != "holochain-0.1.7-rc.0" ]]; then
+              echo "bump.sh failed to find 0.1.7-rc.0: [${RC_VERSION}]"
+              exit 1
+          fi
+          
+          CURRENT_VERSION=$(./scripts/bump-holochain.sh main false | tail -n 1)
```

Comment:
Should there be bumps for version 0.2 and 0.3?

Reply:
Oh good point, let me have another look at this

Reply:
Actually no, they're tested above because this is just about pattern matching

main -> weekly
main-x.y -> holochain x.y.*
main-x.y-rc -> holochain x.y.*-rc-*

Reply:
Although "weekly" was a previous concept and actually the script just knows to look for 0.4.* - that will need to change with each release, or we can pull it out into a file that's more obviously going to need an update after each release or something